src/test/java/com/google/cloud/spanner/pgadapter/JdbcMockServerTest.java

```diff
@@ -405,4 +417,18 @@ public void testDdl() throws SQLException {
     assertEquals(1, updateDatabaseDdlRequests.get(0).getStatementsCount());
     assertEquals(sql, updateDatabaseDdlRequests.get(0).getStatements(0));
   }
+
+  @Test
+  public void testInvalidDdl() throws SQLException {
+    String sql = "CREATE TABLE foo (id int64 primary key)";
+    mockDatabaseAdmin.addException(
```

Comment:
@libingye816 The mock server that we were using for database admin operations (DDL) did not support adding exceptions (and also did a lot of other things that we don't need). This PR replaces that mock server with a much simpler one that also allows us to just add an exception when we want it to fail.